server/src/main/java/org/opensearch/indices/ShardLimitValidator.java

```diff
@@ -261,14 +261,15 @@ static Optional<String> checkShardLimit(
             return Optional.empty();
         }
 
+        int computedMaxShards = (int) Math.min(Integer.MAX_VALUE, (long) maxShardsPerNodeSetting * nodeCount);
         int maxShardsInCluster = maxShardsPerClusterSetting;
         if (maxShardsInCluster == -1) {
-            maxShardsInCluster = maxShardsPerNodeSetting * nodeCount;
+            maxShardsInCluster = computedMaxShards;
         } else {
-            maxShardsInCluster = Math.min(maxShardsInCluster, maxShardsPerNodeSetting * nodeCount);
+            maxShardsInCluster = Math.min(maxShardsInCluster, computedMaxShards);
         }
 
-        int currentOpenShards = state.getMetadata().getTotalOpenIndexShards();
+        long currentOpenShards = state.getMetadata().getTotalOpenIndexShards();
```

Comment:
it can be int only?

Reply:
just want the `testComputedMaxShardsOfClusterIntOverFlow` throwing exception, if currentOpenShards is int type, 
`
if ((currentOpenShards(2) + newShards(Integer.MAX_VALUE - 1)) > maxShardsInCluster(Integer.MAX)) 
` will be false, int overflow occurred, we can't see any exception.

Reply:
@shwetathareja  - The return type of `getTotalOpenIndexShards` is declared to be integer and this is computed by  accumulating the count (integer type) across all the indices in cluster. Instead of changing the local variable `currentOpenShards` to integer we might need to change the return type of `getTotalOpenIndexShards` to return long.

Reply:
@rajiv-kv, The shard count of the cluster cannot reach the Integer.MAX,  so it seems no need to change the return type of getTotalOpenIndexShards.

Reply:
Got it @kkewwei . I think in that case we can leave `currentOpenShards` as int and modify the test.